packages/devextreme/js/__internal/ui/list/m_list.base.ts

```diff
@@ -345,6 +371,15 @@ export const ListBase = CollectionWidget.inherit({
   },
 
   _init() {
+    const { collapsibleGroups } = this.option();
+
+    if (collapsibleGroups) {
+      this._activeStateUnit = [
+        ...DEFAULT_ACTIVE_STATE_UNIT_ARRAY,
+        `.${LIST_GROUP_HEADER_CLASS}`,
+      ].join(',');
```

Comment:
[refactoring]
This can be done though defaultOptionRules mechnism

Reply:
By the way, ‘working defaultOptionRules mechanism’ sounds like a good topic for the documentation section.

Reply:
I'm pretty sure defaultOptionsRules can't help in this scenario, as _activeStateUnit is not a component option, but a class property:

https://github.com/DevExpress/DevExtreme/blob/24_2/packages/devextreme/js/__internal/ui/list/m_list.base.ts#L62 

I'm also pretty sure that defaultOptionsRules ideologically can only be used exclusively for options.

Also, I'm sure that rewriting _activeStateUnit to an option would be difficult within this PR because this property is inherent in many classes (see screenshot below).

Let's make a separate task for this? 

<img width="491" alt="image" src="https://github.com/user-attachments/assets/74500e61-a6fe-4b6d-9857-0ef2c8fc90cc">

Reply:
Yeah, I haven't noticed it's not an option.
I agree to create a separate card for refactoring - this code looks very strange for me, even in terms of naming